src/biogeophys/SnowSnicarMod.F90

```diff
@@ -1372,62 +1797,671 @@ subroutine SnowOptics_init( )
-     call ncd_io( 'ext_cff_mss_ocphob', ext_cff_mss_oc2, 'read', ncid, posNOTonfile=.true.)
-
-     ! dust species 1 Mie parameters
-     call ncd_io( 'ss_alb_dust01', ss_alb_dst1,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust01', asm_prm_dst1,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust01', ext_cff_mss_dst1, 'read', ncid, posNOTonfile=.true.)
-
-     ! dust species 2 Mie parameters
-     call ncd_io( 'ss_alb_dust02', ss_alb_dst2,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust02', asm_prm_dst2,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust02', ext_cff_mss_dst2, 'read', ncid, posNOTonfile=.true.)
-
-     ! dust species 3 Mie parameters
-     call ncd_io( 'ss_alb_dust03', ss_alb_dst3,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust03', asm_prm_dst3,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust03', ext_cff_mss_dst3, 'read', ncid, posNOTonfile=.true.)
-
-     ! dust species 4 Mie parameters
-     call ncd_io( 'ss_alb_dust04', ss_alb_dst4,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust04', asm_prm_dst4,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust04', ext_cff_mss_dst4, 'read', ncid, posNOTonfile=.true.)
-
+     !--------------------- for 5-band data
+     select case (snicar_numrad_snw)
+     case (5)  ! 5-band case
+
+        select case (snicar_solarspec)
+        ! mid-latitude winter spectrum
+        case ('mid_latitude_winter')
+           ! flux weights/spectrum
+           call ncd_io( 'flx_wgt_dir5_mlw', flx_wgt_dir,     'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'flx_wgt_dif5_mlw', flx_wgt_dif,     'read', ncid, posNOTonfile=.true.)
+           !
+           ! THIS NOTE APPLIES TO ALL THE call ncd_io LINES BELOW WHERE
+           ! bcphob AND ocphob GET ASSIGNED TO VARIABLES SUFFIXED bc1/oc1:
+           !
+           ! Assumption (1) applies here, in the input section.
+           ! Assumption (2) applies later, in the snicar code.
+           !
+           ! 1) In this section, hydrophillic particles behave like hydrophobic
+           ! particles. We assume bc1/oc1 to have the same optics as bc2/oc2
+           ! because sulfate coating on the bc1/oc1 surface is assumed to be
+           ! dissolved into the hydrometeo (i.e, snow grain here) during the
+           ! deposition process. This is different from the assumption made in
+           ! prior model versions, where bc1/oc1 was coated by undissolved
+           ! sulfate.
+           ! 2) Later, in the snicar code, if the bc-snow internal mixing option
+           ! is on, bc1/oc1 (internally mixed within the snow grain) will be
+           ! treated differently than bc2/oc2 (mixed externally or outside the
+           ! snow grain).
```

Comment:
@cenlinhe this is the new comment that I added here to prevent confusion. I'm open to suggestions.

Reply:
this comment looks good to me. thank you!